gpusim.cpp

```diff
@@ -40,6 +42,43 @@ const int DATABASE_VERSION = 3;
 
 namespace gpusim
 {
+
+class DecompressAssignFPRunnable : public QRunnable {
+public:
+    QByteArray compressed_data;
+    vector<char>& fp_vector;
+
+    DecompressAssignFPRunnable(vector<char>& fp_in) : fp_vector(fp_in) {};
+    void run() override {
+        QByteArray fp_qba = qUncompress(compressed_data);
+        compressed_data.clear();
+        fp_vector.reserve(fp_qba.size());
+        fp_vector.insert(fp_vector.begin(), fp_qba.data(), fp_qba.data()+fp_qba.size());
```

Comment:
Maybe I'm missing nuances of QRunnables, but it seems weird that you init with output memory, and have input as a public member. Why not make both args to the constructor, and have compressed_data private?

Reply:
Near-immediately when the QRunnable is done executing, the QThreadPool deletes it, so it's not really a persistent object - that's why it makes sense to have this object write to a persistent space.